core/common/src/main/scala/org/typelevel/otel4s/context/propagation/ContextPropagators.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.context.propagation
+
+/** A container of the registered propagators for every supported format.
+  *
+  * @tparam Ctx
+  *   the type of the Context
+  */
+trait ContextPropagators[Ctx] {
+
+  /** Returns a text map propagator to extract or inject data.
+    */
+  def textMapPropagator: TextMapPropagator[Ctx]
+}
+
+object ContextPropagators {
+
+  /** Creates a [[ContextPropagators]] which can be used to extract and inject
+    * context in text payloads with the given [[TextMapPropagator]].
+    *
+    * '''Hint''': use [[TextMapPropagator.composite]] to combine multiple text
+    * map propagators.
+    *
+    * @example
+    *   {{{
+    * val w3cPropagator: TextMapPropagator[Context] = ???
+    * val httpTracePropagator: TextMapPropagator[Context] = ???
+    * val textMapPropagator = TextMapPropagator.composite(w3cPropagator, httpTracePropagator)
+    * val contextPropagators = ContextPropagators.create(textMapPropagator)
+    *   }}}
+    *
+    * @param textMapPropagator
+    *   the text map propagator to extract or inject data
+    *
+    * @tparam Ctx
+    *   the context to use to extra or inject data
+    */
+  def create[Ctx](
+      textMapPropagator: TextMapPropagator[Ctx]
```

Comment:
I think this would be better named `apply`. also, you could make it varargs and have it delegate that part to `TextMapPropagator.composite`, which seems tidy

```suggestion
  def apply[Ctx](
      textMapPropagators: TextMapPropagator[Ctx]*
```

Reply:
I tend to agree that `create` is, uhm, verbose and java-ish. `apply` would fit better.